app/lib/attachment_batch.rb

```diff
@@ -95,13 +95,11 @@ def remove_files
     # objects can be processed at once, so we have to potentially
     # separate them into multiple calls.
 
-    keys.each_slice(LIMIT) do |keys_slice|
-      logger.debug { "Deleting #{keys_slice.size} objects" }
-
-      bucket.delete_objects(delete: {
-        objects: keys_slice.map { |key| { key: key } },
-        quiet: true,
-      })
+    # Mozilla Social - delete individually instead of in batches,
+    # since GCP XML API doesn't support batch delete
+    logger.debug { "Deleting #{keys.size} objects" }
+    keys.each do |key|
+      bucket.object(key).delete
```

Comment:
interesting that your non-batch version is less lines and less complex than the original batch version that needs to be sliced anyway! 🚀

Reply:
haha! the batching makes a difference on the S3 side - putting it into slices of 1000 means all those 1000 media get deleted simultaneously, rather than O(n) with the non-batch way.

but it all gets deleted eventually so it shouldn't be an issue!